erpnext/patches/v7_1/fix_start_end_date_salary_register.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from calendar import monthrange
+
+def execute():
+	sql = """select employee, month, fiscal_year, status, employee_name, name
+		from `tabSalary Slip`
+		where start_date is null
+		and end_date is null"""
+
+	for date in frappe.db.sql(sql, as_dict=True):
+		years = date.fiscal_year.split("-")
+		if int(date.month) >= 7:
+			year = years[0]
+		else:
+			year = years[1]
+		start_date = year + "-" + date.month + "-01"
+		end_date = year + "-" + date.month + "-" + str(monthrange(int(year), int(date.month))[1])
+		if date.docstatus == 0:
+			status = "Draft"
+		elif date.docstatus == 1:
+			status = "Submitted"
+		else:
+			status = "Cancelled"
+
+		frappe.db.sql("""update `tabSalary Slip`
+			set start_date = %(start_date)s, end_date = %(end_date)s, status = %(status)s,
+				salary_structure = %(salary_structure)s
```

Comment:
why set salary structure things could of changed since creation 
and why are u changing status's
seems dangerous